src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -0,0 +1,311 @@
+    // Want to check that when code is created or updated, you check if it is a service account and turn on properties
+    // For example: Creator/owner of service account (extension or user)
+    // All end-to-end tests can be written without extensions and we can test everything
+    // Should not need to test create, update, list, get as unit tests but will want to test service account creation and deletion
+    // I.e. service account is invalid because no owner -- can leave tests till later.
+
+
+    @Inject
+    public UserService(
+            ClusterService clusterService,
+            ConfigurationRepository configurationRepository,
+            Settings settings,
+            Client client
+    ) {
+        this.clusterService = clusterService;
+        this.configurationRepository = configurationRepository;
+        this.securityIndex = settings.get(ConfigConstants.SECURITY_CONFIG_INDEX_NAME,
+                ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
+        this.client = client;
+    }
+
+    /**
+     * Load data for a given CType
+     * @param config CType whose data is to be loaded in-memory
+     * @return configuration loaded with given CType data
+     */
+    protected static final SecurityDynamicConfiguration<?> load(final CType config, boolean logComplianceEvent) {
+        SecurityDynamicConfiguration<?> loaded = configurationRepository.getConfigurationsFromIndex(Collections.singleton(config), logComplianceEvent).get(config).deepClone();
+        return DynamicConfigFactory.addStatics(loaded);
+    }
+
+    protected void saveAndUpdateConfiguration(final Client client, final CType cType,
+                                              final SecurityDynamicConfiguration<?> configuration) {
+        final IndexRequest ir = new IndexRequest(this.securityIndex);
+
+        // final String type = "_doc";
+        final String id = cType.toLCString();
+
+        configuration.removeStatic();
+
+        try {
+            client.index(ir.id(id)
+                            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                            .setIfSeqNo(configuration.getSeqNo())
+                            .setIfPrimaryTerm(configuration.getPrimaryTerm())
+                            .source(id, XContentHelper.toXContent(configuration, XContentType.JSON, false)));
+        } catch (IOException e) {
+            throw ExceptionsHelper.convertToOpenSearchException(e);
+        }
+    }
+
+    /**
+     * This function will handle the creation or update of a user account.
+     *
+     * @param accountDetailsAsString A string JSON of different account configurations.
+     * @return
+     * @throws ServiceAccountRegistrationException
+     * @throws IOException
+     */
+    public SecurityDynamicConfiguration<?> createOrUpdateAccount(String accountDetailsAsString) throws IOException, AccountCreateOrUpdateException, ServiceAccountRegistrationException {
```

Comment:
I think this needs a refactor to be more maintainable, different validations can be moved into functions to make this more straight forward.

Afterward - please add unit test to verify this functionality